packages/abtest/src/utils.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CONTROL,
+	EXPERIMENT,
+	OPTION_NAME,
+	WC_ADMIN_NAMESPACE,
+} from './constants';
+
+/**
+ * Get named key from localStorage.
+ *
+ * @param {string} name Key name.
+ *
+ * @return {string} - Value of named key.
+ */
+export const getCachedGroup = ( name ) => window.localStorage.getItem( name );
+
+/**
+ * Set localStorage named key to value.
+ *
+ * @param {string} name
+ * @param {string} value
+ */
+export const setCachedGroup = ( name, value ) =>
+	window.localStorage.setItem( name, value );
+
+/**
+ * Fetch abtest option from backend.
+ *
+ * @param {string} name
+ */
+const getABTestOption = ( name ) => {
+	return apiFetch( {
+		path: `${ WC_ADMIN_NAMESPACE }/options?options=${ OPTION_NAME }_${ name }`,
+	} );
+};
+
+/**
+ * Set abtest option in backend.
+ *
+ * @param {string} name
+ * @param {string} value
+ */
+const setABTestOption = ( name, value ) => {
+	apiFetch( {
+		method: 'POST',
+		path: `${ WC_ADMIN_NAMESPACE }/options`,
+		data: { [ `${ OPTION_NAME }_${ name }` ]: value },
+	} );
+};
+
+/**
+ * Fetch option from backend and set localStorage accordingly.
+ *
+ * If option doesn't exist, then pick a group, update in the backend,
+ * set localStorage, and trigger a tracks event.
+ *
+ * If apiFetch fails, default to control.
+ *
+ * @param {string} name
+ *
+ * @return {string} - Group name (CONTROL or EXPERIMENT).
+ */
+export const getAndSetGroup = async ( name ) => {
+	try {
+		const option = await getABTestOption( name );
+		const group =
+			option[ `${ OPTION_NAME }_${ name }` ] || getRandomGroup();
+
+		if ( ! option[ `${ OPTION_NAME }_${ name }` ] ) {
+			setABTestOption( name, group );
+		}
+
+		setCachedGroup( name, group );
+		recordABTestEvent( name, group, 'persisted' );
+		return group;
+	} catch {
+		setCachedGroup( name, CONTROL );
+		recordABTestEvent( name, CONTROL, 'fetch_failed' );
+		return CONTROL;
+	}
+};
+
+/**
+ * Randomly select between CONTROL and EXPERIMENT groups.
+ */
+export const getRandomGroup = () =>
+	Math.random() < 0.5 ? CONTROL : EXPERIMENT;
```

Comment:
How about extracting the `0.5` value to a prop on `<ABTest />`? There may be a desire for a conservative approach of showing the experiment to only 10% or something

Reply:
I had considered this initially but based on our needs for this package, decided not to. Now that you point it out though, it does make more sense to expose this as a prop. I've made the change in https://github.com/woocommerce/woocommerce-admin/pull/6536/commits/adf2d654e07132f35f82b787b67a7fdfb995b3b7